Update the data in the showAttractions

shirazyomtov opened the pull request:
1.Update the context
2.Update the showAttractions
3.Add the addNewAttraction in the AttractionsForm

shirazyomtov committed:
Update the data in the showAttractions

1.Update the context
2.Update the showAttractions
3.Add the addNewAttraction in the AttractionsForm

irisri commented on client/src/context/AttractionContext.js:
```diff
@@ -18,7 +18,7 @@ const AttractionProvider = ({ children }) => {
     
-    const getAttractions = () => {
+    const getFilteredAttractions = () => {
         const res = attractions.filter((attraction) => {
```
just return you don't need to do res

shirazyomtov replied:
Done

irisri commented on client/src/context/AttractionContext.js:
```diff
@@ -37,8 +37,22 @@ const AttractionProvider = ({ children }) => {
+    }
+
+    const updateAttraction = (updatedAttraction) => {
+        const updatedAttractions = attractions.map((attraction) => {
```
Great!

irisri commented on client/src/components/UploadData/AttractionsForm.js:
```diff
@@ -45,7 +47,7 @@ const AttractionForm = () => {
         attraction
       );
-
+      addNewAttraction(attraction);
```
delete the console log in line 43 and in 55

irisri replied:
please add a ticket intrello to move all http calls to a service file

shirazyomtov replied:
The unnecessary lines have been removed and a new ticket has been added to Trello for moving all HTTP calls to a service file

irisri commented on client/src/components/Attractions/ShowAttractions/ShowAttractions.js:
```diff
@@ -28,7 +28,7 @@ const ShowAttractions = () => {
     const fetchAttractions = async () => {
```
I dont think that the useEffect is necessary, the component will re-render when you update the filters (because it's in useState in the context) so you can just use the getFilteredAttractions in line 15 and update it to const.
and you don't need the try catch - it's not an HTTP request.

shirazyomtov replied:
Yes, it is unecessary, I removed the useEffect.

shirazyomtov committed:
Update the files

1.Remove the useEffect in ShowAttractions
2.Update the AttractionsContext
3.Update the AttractionsForm

shirazyomtov committed:
Change the varible currAttractions to const

irisri commented on client/src/components/Attractions/ShowAttractions/ShowAttractions.js:
```diff
@@ -23,29 +22,9 @@ const ShowAttractions = () => {
   // console.log('currentRecords changed ' + JSON.stringify(currAttractions, null, 2));
```
delete the console.log please

irisri reviewed (approved):
looks so much cleaner! Great job

shirazyomtov committed:
Merge branch 'main' into search-bar
